cdc/sink/dmlsink/txn/worker.go

```diff
@@ -122,27 +117,38 @@ func (w *worker) runLoop() error {
 				zap.Int("workerID", w.ID))
 			return nil
 		case txn := <-w.txnCh.Out():
+			// we get the data from txnCh.out until no more data here or reach the state that can be flushed.
+			// If no more data in txnCh.out, and also not reach the state that can be flushed,
+			// we will wait for 10ms and then do flush to avoid too much flush with small amount of txns.
 			if txn.txnEvent != nil {
 				needFlush = w.onEvent(txn)
+				if !needFlush {
+					delay := time.After(w.flushInterval)
+					for !needFlush {
+						select {
+						case txn := <-w.txnCh.Out():
+							needFlush = w.onEvent(txn)
+						case <-delay:
+							needFlush = true
+						}
+					}
```

Comment:
If it ends the `if` block starts at line `125`, I think we can flush here directly.

Reply:
Do you mean that we can move L139-L151 to L135 ？

Reply:
Yes.